handlers.py

```diff
@@ -59,69 +64,71 @@ async def registration(message: types.Message):
+    async def adding_to_db(message: types.Message):
         try:
-            database.register(f"@{message.from_user.username}", data['name'], data['surname'], message.from_user.id)
-            await message.answer('Vielen Dank fur die Registrierung!')
+            database.register(f"@{message.from_user.username}", message.from_user.first_name, message.from_user.id, message.contact['phone_number'])
+            await bot.send_message(chat_id=message.from_user.id, text="Thank you for the registration!")
             await bot.send_message(USER_ID,
-                                   text=f"New user: {' '.join([data['name'], data['surname'], f'@{message.from_user.username}'])}")
+                                   text=f"New user: {' '.join([message.from_user.first_name, f'@{message.from_user.username}', message.contact['phone_number']])}")
         except sqlite3.IntegrityError:
-            await message.answer("You've already been registered!")
-    await state.finish()
+            await bot.send_message(chat_id=message.from_user.id, text="You've already been registered!")
 
 
 @dp.message_handler(lambda message: message.text == "Track a person")
 async def track_person(message: types.Message):
-    await message.answer("Please enter username of a person (with @ symbol)")
-
-@dp.message_handler()
-async def find_person(message: types.Message):
-    tracking_id.append(message.from_user.id)
-    tracked_username = message.text
-    try:
-        assert '@' in message.text
-    except AssertionError:
-        await bot.send_message(message.from_user.id, text="Please enter username WITH '@'")
-    result = database.tracked_id(tracked_username)
-    Person.tracked_id = result[0][0]
-    #print(tracking_id, Person.tracked_id)
-    keyboard = types.ReplyKeyboardMarkup(resize_keyboard=True, one_time_keyboard=True)
-    button1 = types.KeyboardButton("Yes", request_location=True)
-    button2 = types.KeyboardButton("No", request_contact=True)
-    keyboard.add(button1, button2)
-    await bot.send_message(Person.tracked_id,
-                           text=f"User @{message.from_user.username} wants to track you, are you agree?",
-                           reply_markup=keyboard)
+    await message.answer("Please enter phone number of a person (choose from your contacts)")
+
+    @dp.message_handler(content_types=['contact'])
+    async def find_person(message: types.Message):
+        print(message.contact['phone_number'][1::])
+        queries[database.tracked_id(message.contact['phone_number'][1::])[0][0]].append(message.from_user.id)
+        print(queries)
+        keyboard = types.ReplyKeyboardMarkup(resize_keyboard=True, one_time_keyboard=True)
+        button1 = types.KeyboardButton("Yes", request_location=True)
+        button2 = types.KeyboardButton("No")
+        keyboard.add(button1, button2)
+        await bot.send_message(database.tracked_id(message.contact['phone_number'][1::])[0][0],
```

Comment:
inside this method you have repetitive construction `message.contact['phone_number'][1::]` - why not to define this variable once in the beggining of method and then to reuse? and also it's not clear why you make slicing by `[1::]`, better to leave a comment upon it

Reply:
every phone numbers starts with "+", that why we need to get rid of it. and it's highly risky to store this in a variable cause it renews every time a new users uses it and perhaps users will click on the button simultaneously